src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +128,2943 @@ const ActorInit Boss_Mo_InitVars = {
+            Math_ApproachF(&this->tentPulse, 0.3f, 0.5f, 0.03f);
+            if ((this->mashCounter >= 40) || (this->timers[0] == 0)) {
+                tentXrot = this->tentRot[15].x;
+                if ((tentXrot < 0) && (sp1B4 >= 0)) {
+                    this->actionState = MO_TENT_RETREAT;
+                    this->invincibilityTimer = 50;
+                    if (&this->actor == player->actor.parent) {
+                        player->unk_850 = 0x65;
+                        player->actor.parent = NULL;
+                        player->csMode = 0;
+                        if (this->timers[0] == 0) {
+                            func_8002F6D4(globalCtx, &this->actor, 20.0f, this->actor.shape.rot.y + 0x8000, 10.0f, 0);
+                        }
+                    }
+                    this->timers[0] = 75;
+                }
+            }
+            if (this->csCamera != 0) {
+                sp138.x = 0;
+                sp138.y = 100.0f;
+                sp138.z = 200.0f;
+                this->cameraYaw -= this->cameraYawRate;
+                Math_ApproachF(&this->cameraYawRate, 0.01, 1.0f, 0.002f);
+                Matrix_RotateY(this->cameraYaw, MTXMODE_NEW);
+                Matrix_MultVec3f(&sp138, &sp12C);
+                Math_ApproachF(&this->cameraEye.x, this->actor.world.pos.x + sp12C.x, 0.1f, 10.0f);
+                Math_ApproachF(&this->cameraEye.y, this->actor.world.pos.y + sp12C.y, 0.1f, 10.0f);
+                Math_ApproachF(&this->cameraEye.z, this->actor.world.pos.z + sp12C.z, 0.1f, 10.0f);
+                Math_ApproachF(&this->cameraAt.x, player->actor.world.pos.x, 0.5f, 50.0f);
+                Math_ApproachF(&this->cameraAt.y, player->actor.world.pos.y, 0.5f, 50.0f);
+                Math_ApproachF(&this->cameraAt.z, player->actor.world.pos.z, 0.5f, 50.0f);
+                Gameplay_CameraSetAtEye(globalCtx, this->csCamera, &this->cameraAt, &this->cameraEye);
+            }
+            break;
+        case MO_TENT_CUT:
+            func_80078914(&this->tentTipPos, NA_SE_EV_WATER_WALL - SFX_FLAG);
+            if (&this->actor == player->actor.parent) {
+                player->unk_850 = 0x65;
+                player->actor.parent = NULL;
+                player->csMode = 0;
+            }
+            Math_ApproachF(&this->tentRippleSize, 0.15f, 0.5f, 0.01);
+            if (this->meltIndex < 41) {
+                for (indS0 = 0; indS0 < 10; indS0++) {
+                    sp120 = this->tentPos[this->meltIndex];
+                    sp120.x += Rand_CenteredFloat(30.0f);
+                    sp120.y += Rand_CenteredFloat(30.0f);
+                    sp120.z += Rand_CenteredFloat(30.0f);
+                    BossMo_SpawnStillDroplet(globalCtx->specialEffects, &sp120, Rand_ZeroFloat(0.1f) + .2f);
+                }
+                this->meltIndex++;
+            }
+            Math_ApproachF(&this->cutScale, 0.0, 1.0f, 0.2f);
+            if ((this->meltIndex >= 41) || (this->timers[0] == 0)) {
```

Comment:
> 40 ? again not sure on context, just seems like a nicer number to me lol

Reply:
lol github ate my greater than sign
either way tho i figured out later on that 41 is the number of core parts right. in which case disregard